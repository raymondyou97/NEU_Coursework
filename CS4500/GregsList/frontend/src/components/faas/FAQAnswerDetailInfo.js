import React from 'react';
import PropTypes from 'prop-types';

const FAQAnswerDetailInfo = ({ faqAnswer, faqAnswers, onChange }) => (
  <div>
    <div>
      <h3>FAQ Answer Details</h3>
      <label>
        ID:
        <select
          value={faqAnswer.id}
          onChange={e => onChange(e.target.value)}
          className="form-control"
        >
          {faqAnswers.map(faa => (
            <option value={faa.id} key={faa.id}>
              {faa.id}
            </option>
          ))}
        </select>
      </label>
      <div>
        <div>
          <label>
            Question:
            <input
              onChange={() => {}}
              className="form-control"
              value={faqAnswer.question}
            />
          </label>
        </div>
        <div>
          <label>
            Answer:
            <input
              onChange={() => {}}
              className="form-control"
              value={faqAnswer.answer}
            />
          </label>
        </div>
      </div>
    </div>
  </div>
);

FAQAnswerDetailInfo.propTypes = {
  faqAnswer: PropTypes.shape({
    id: PropTypes.number.isRequired,
    question: PropTypes.string.isRequired,
    answer: PropTypes.string.isRequired,
  }).isRequired,
};

export default FAQAnswerDetailInfo;
