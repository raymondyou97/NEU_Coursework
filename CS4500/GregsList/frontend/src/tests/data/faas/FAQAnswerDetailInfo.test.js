import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import FAQAnswerDetailInfo from '../../../components/faas/FAQAnswerDetailInfo';
import { findAllFAQAnswers } from './MockFaaService';

test('FAQ Answer Detail Components', () => {
  const faas = findAllFAQAnswers();

  const component = shallow(
    <FAQAnswerDetailInfo
      faqAnswer={faas[0]}
      faqAnswers={faas}
      onChange={() => {}}
    />
  );
  expect(component.find('option').length).toEqual(faas.length);

  const titleValue = component.find('input');
  expect(titleValue.at(0).props().value).toEqual(faas[0].question);
  expect(titleValue.at(1).props().value).toEqual(faas[0].answer);
});

test('Snapshot renders', () => {
  const faas = findAllFAQAnswers();

  const component = renderer.create(
    <FAQAnswerDetailInfo
      faqAnswer={faas[0]}
      faqAnswers={faas}
      onChange={() => {}}
    />
  );

  let componentTree = component.toJSON();
  expect(componentTree).toMatchSnapshot();
});
