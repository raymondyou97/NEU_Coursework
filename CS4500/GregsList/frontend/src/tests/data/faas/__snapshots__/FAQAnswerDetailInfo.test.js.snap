// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot renders 1`] = `
<div>
  <div>
    <h3>
      FAQ Answer Details
    </h3>
    <label>
      ID:
      <select
        className="form-control"
        onChange={[Function]}
        value={1}
      >
        <option
          value={1}
        >
          1
        </option>
        <option
          value={2}
        >
          2
        </option>
        <option
          value={3}
        >
          3
        </option>
        <option
          value={4}
        >
          4
        </option>
        <option
          value={5}
        >
          5
        </option>
      </select>
    </label>
    <div>
      <div>
        <label>
          Question:
          <input
            className="form-control"
            onChange={[Function]}
            value="how many employees do you have?"
          />
        </label>
      </div>
      <div>
        <label>
          Answer:
          <input
            className="form-control"
            onChange={[Function]}
            value="we have 300 people."
          />
        </label>
      </div>
    </div>
  </div>
</div>
`;
